### ISSUE TITLE
Memory issue when plotting large arrays with pcolormesh

### ISSUE THREAD

**Issue by dkillick** [matplotlib/matplotlib#3095]
Plotting large-ish arrays using pcolormesh with current (e2c5918) matplotlib master looks like it might have a memory leak. Running the following code on matplotlib v1.3.1 produces a `QuadMesh` in approximately a second. However running the same code on matplotlib master floods the memory until either the pcolormesh call is aborted or it runs out of memory.

```python
import matplotlib.pyplot as plt
import numpy as np

data = np.random.rand(3000*3000).reshape(3000, 3000)
plt.pcolormesh(data)
plt.show()
```

**Comment by tacaswell** [comment]
It looks like this came from 88b722f4748b84007e87c5ec8d0cf7c262a7958b which removed an over-loading of `get_datalim` on `Quadmesh`.  This causes the `get_windowextent` method from collections to try to make all of the paths for the quadmesh _as python path objects_ which causes at least a 5x blow up in the memory used (just from the _data_, let alone the `Path` objects).  Working on a PR to more efficiently determine the extent (and getting very confused with using offsets with `QuadMesh`).

**Comment by tacaswell** [comment]
closed by #4045
